server/src/test/java/org/elasticsearch/search/geo/GeoBoundsGenericWriteableTests.java

```diff
@@ -0,0 +1,120 @@
+    }
+
+    @Override
+    protected GeoBoundingBox mutateInstance(GenericNamedWriteable instance) throws IOException {
+        assert instance instanceof GeoBoundingBox : "Expected GeoBoundingBox";
+        GeoBoundingBox geoBoundingBox = (GeoBoundingBox) instance;
+        double width = geoBoundingBox.right() - geoBoundingBox.left();
+        double height = geoBoundingBox.top() - geoBoundingBox.bottom();
+        double top = geoBoundingBox.top() - height / 4;
+        double left = geoBoundingBox.left() + width / 4;
+        double bottom = geoBoundingBox.bottom() + height / 4;
+        double right = geoBoundingBox.right() - width / 4;
+        return new GeoBoundingBox(new GeoPoint(top, left), new GeoPoint(bottom, right));
+    }
+
+    @Override
+    protected GenericNamedWriteable copyInstance(GenericNamedWriteable original, TransportVersion version) throws IOException {
+        return copyInstance(original, version, version);
+    }
+
+    /**
+     * Read and write with different serialization versions
+     */
+    protected GenericNamedWriteable copyInstance(
+        GenericNamedWriteable original,
+        TransportVersion writeVersion,
+        TransportVersion readVersion
+    ) throws IOException {
+        try (BytesStreamOutput output = new BytesStreamOutput()) {
+            output.setTransportVersion(writeVersion);
+            output.writeGenericValue(original);
+            try (StreamInput in = new NamedWriteableAwareStreamInput(output.bytes().streamInput(), getNamedWriteableRegistry())) {
+                in.setTransportVersion(readVersion);
+                return readGenericValue(in);
+            }
+        }
+    }
+
+    private GenericNamedWriteable readGenericValue(StreamInput in) throws IOException {
+        Object obj = in.readGenericValue();
+        if (obj instanceof GenericNamedWriteable result) {
+            return result;
+        }
+        throw new IllegalStateException("Read result of wrong type: " + obj.getClass().getSimpleName());
+    }
+
+    @Override
+    protected NamedWriteableRegistry getNamedWriteableRegistry() {
+        return registry;
+    }
+
+    @Override
+    protected Class<GenericNamedWriteable> categoryClass() {
+        return GenericNamedWriteable.class;
+    }
+
+    public void testSerializationFailsWithOlderVersion() throws IOException {
+        TransportVersion valid = TransportVersion.V_8_500_070;
+        TransportVersion older = TransportVersion.V_8_500_069;
+        for (int runs = 0; runs < NUMBER_OF_TEST_RUNS; runs++) {
+            GenericNamedWriteable testInstance = createTestInstance();
+
+            // Test reading and writing with valid versions
+            GenericNamedWriteable deserializedInstance = copyInstance(testInstance, valid, valid);
+            assertEqualInstances(testInstance, deserializedInstance);
+
+            // Test writing fails with older version
+            try {
+                copyInstance(testInstance, older, valid);
```

Comment:
I think we should just verify that writing to an older stream fails, no need to make a `copyInstance` which tries to read it back again.

Reply:
OK. After simplifying the test I replaced this with just the writing code.